### PR TITLE
Onboard system authorization core

### PR DESCRIPTION
**Pull request by senthalan**
### Purpose
This pull request introduces mechanisms to explicitly mark and identify contexts where security enforcement is intentionally skipped or where a request is considered an internal runtime operation. It also adds comprehensive tests for these flows and refactors the handling of public paths and skip-security logic in the security service. Additionally, a new `sysauthz` package is introduced for system-level authorization constants and models.

### High-Level Design

```mermaid
graph TD
    subgraph "Tier 1 — HTTP Security Middleware"
        A["security.Process(request)"] --> B{"Public path?"}
        B -- Yes --> C["WithRuntimeContext(ctx)"]
        B -- No --> D{"Authenticator found?"}
        D -- Yes --> E["Authenticate → SecurityContext"]
        D -- No --> F{"SKIP_SECURITY?"}
        E --> G["authorize(permissions check)"]
        G -- Pass --> H["Enriched ctx returned"]
        G -- Fail --> F
        F -- Yes --> I["withSecuritySkipped(ctx)"]
        F -- No --> J["Error returned"]
    end

    subgraph "Tier 2 — Domain-Level Authorization"
        K["sysauthz.IsActionAllowed(ctx, action, actionCtx)"] --> L{"SecuritySkipped?"}
        L -- Yes --> M["✅ Allow"]
        L -- No --> N{"RuntimeContext?"}
        N -- Yes --> O["✅ Allow"]
        N -- No --> P{"Authenticated?"}
        P -- No --> Q["❌ Deny"]
        P -- Yes --> R{"'system' permission?"}
        R -- Yes --> S["✅ Allow"]
        R -- No --> T["Check action-specific permissions"]
        T -- Pass --> U["Evaluate policy chain"]
        T -- Fail --> V["❌ Deny"]
        U -- Pass --> W["✅ Allow"]
        U -- Fail --> X["❌ Deny"]
    end

    H --> K
    C --> K
    I --> K
```

**Security context enhancements:**

* Added `withSecuritySkipped` and `IsSecuritySkipped` functions to mark and detect when security enforcement is skipped, along with corresponding context key (`securitySkippedKey`). This is used for cases where security checks are intentionally bypassed, such as when `THUNDER_SKIP_SECURITY` is enabled. [[1]](diffhunk://#diff-3eb3fe67693bcafc41baf4781dbaa092bca2497f3980116b545cb418ec68b345R30-R35) [[2]](diffhunk://#diff-3eb3fe67693bcafc41baf4781dbaa092bca2497f3980116b545cb418ec68b345R67-R84)
* Introduced `WithRuntimeContext` and `IsRuntimeContext` functions to mark and detect internal runtime callers, with a dedicated context key (`runtimeContextKey`). This allows the system to grant a limited set of actions for internal operations that do not have a user token. [[1]](diffhunk://#diff-3eb3fe67693bcafc41baf4781dbaa092bca2497f3980116b545cb418ec68b345R30-R35) [[2]](diffhunk://#diff-3eb3fe67693bcafc41baf4781dbaa092bca2497f3980116b545cb418ec68b345R153-R174)

**Security service logic updates:**

* Updated the security service (`handleAuthError`) to use `WithRuntimeContext` for public paths and `withSecuritySkipped` for skip-security scenarios, ensuring the context is correctly marked for downstream authorization logic.

**Testing improvements:**

* Added and expanded tests for `IsSecuritySkipped` and `WithRuntimeContext` to ensure correct behavior in all edge cases, including nil and enriched contexts. [[1]](diffhunk://#diff-cbde0252501132762bee058fd21a53c1f049db345fa617eb7e76023682ee73eaR417-R481) [[2]](diffhunk://#diff-cbde0252501132762bee058fd21a53c1f049db345fa617eb7e76023682ee73eaR521-R564)
* Updated security service tests to assert the presence of runtime or skip-security markers in the context, replacing previous checks that only compared context equality. [[1]](diffhunk://#diff-26ef0ee1fff32423b261939280c08043cbd7389677b4208a68db8f1143086d0bL127-R128) [[2]](diffhunk://#diff-26ef0ee1fff32423b261939280c08043cbd7389677b4208a68db8f1143086d0bL424-R426) [[3]](diffhunk://#diff-26ef0ee1fff32423b261939280c08043cbd7389677b4208a68db8f1143086d0bL574-R577) [[4]](diffhunk://#diff-26ef0ee1fff32423b261939280c08043cbd7389677b4208a68db8f1143086d0bL598-R603) [[5]](diffhunk://#diff-26ef0ee1fff32423b261939280c08043cbd7389677b4208a68db8f1143086d0bL652-R658) [[6]](diffhunk://#diff-26ef0ee1fff32423b261939280c08043cbd7389677b4208a68db8f1143086d0bL678-R704)

**System authorization package:**

* Introduced a new `sysauthz` package containing:
  - Constants for resource types, actions, and permissions.
  - Data models for resource types, actions, action context, and accessible resources.
  - A package initializer for the system authorization service interface. [[1]](diffhunk://#diff-6234d2de182f088cb70112c1f47ab27327943f68c7d8a150aeeec08951735d34R1-R81) [[2]](diffhunk://#diff-70e30399a26782e3a28ee2aae02073be29dd1a25c47bb065f50931ab80e42df2R1-R25) [[3]](diffhunk://#diff-ea519a9269ac7fca0c61a2aaf82894fadb7d4b7b14483df3998cd40c14ef949eR1-R50)

### Related Issues
- Fix https://github.com/asgardeo/thunder/issues/1501

### Related PRs
- N/A

### Checklist
- [x] Followed the contribution guidelines.
- [x] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
    - [ ] Ran Vale and fixed all errors and warnings
- [ ] Tests provided. (Add links if there are any)
    - [ ] Unit Tests
    - [ ] Integration Tests
- [x] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [x] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

## Summary by CodeRabbit

* **New Features**
  * Added a system-level authorization service with action- and resource-aware access control.
  * Introduced a hierarchical, granular permission model for organization units, users, and groups.
  * Enabled action-level enforcement for create/read/update/delete/list operations.
  * Added markers to identify internal runtime calls and to allow skipping authorization for internal operations.

* **Tests**
  * Extensive tests covering permission resolution, API permission mapping, authorization decisions, and policy behaviors.

### CODE REVIEW

File: backend/internal/system/security/permissions.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+// HasSufficientPermission returns true if any permission in userPermissions satisfies
+// the required permission using hierarchical scope matching.
+//
+// Matching rules:
+//   - Empty required: always satisfied (self-service paths with no specific permission requirement)
+//   - Exact match: "system:ou:view" satisfies "system:ou:view"
+//   - Parent scope: "system:ou" satisfies "system:ou:view" (parent covers all children)
+//   - Root scope: "system" satisfies any "system:*" permission
+func HasSufficientPermission(userPermissions []string, required string) bool {
```
Comment (rajithacharith) [inline backend/internal/system/security/permissions.go:224]: Do we need to export this?

Comment (rajithacharith) [reply backend/internal/system/security/permissions.go:224]: Same for HasSystemPermission

File: backend/internal/system/sysauthz/service.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+			log.String("subject", log.MaskString(subject)))
+	}
+
+	return true, nil
+}
+
+// GetAccessibleResources returns the set of resources the caller can access for the given
+// action and resource type.
+//
+// Decision logic (in order):
```
Comment (rajithacharith) [inline backend/internal/system/sysauthz/service.go]: Do we need these logic comments. 
There are comments inside the function as well

File: backend/internal/system/sysauthz/service.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+			log.String("subject", log.MaskString(subject)),
+			log.Int("accessibleCount", len(result.IDs)))
+	}
+	return result, nil
+}
+
+// Permission resolution and matching logic lives in the security package:
+// - security.ResolveActionPermission(action) — maps Action → required permission
+// - security.HasSystemPermission(permissions) — fast-path check for root scope
+// - security.HasSufficientPermission(permissions, required) — hierarchical scope match
```
Comment (rajithacharith) [inline backend/internal/system/sysauthz/service.go]: Is this the end of the file?

File: backend/internal/system/security/permissions.go (hunk 1 above)
Comment (senthalan) [reply backend/internal/system/security/permissions.go:224]: These need to be accessed by the `sysauthz` package
